build-tools/packages/build-tools/src/common/gitRepo.ts

```diff
@@ -226,33 +226,58 @@ export class GitRepo {
 	}
 
 	/**
-	 * Returns an array containing repo repo-relative paths to all the files in the provided directory.
-	 * A given path will only be included once in the array; that is, there will be no duplicate paths.
-	 * Note that this function excludes files that are deleted locally whether the deletion is staged or not.
+	 * Returns git output of all non-ignored files including deleted files that are not staged.
+	 * A given path may be included multiple timed in the array if git version does not support
+	 * `--deduplicate` option (introduced in v2.31.0).
+	 * Note that this function excludes files that are deleted locally AND staged.
 	 *
 	 * @param directory - A directory to filter the results by. Only files under this directory will be returned. To
```

Comment:
not: new param is undocumented.

Reply:
I left it out because it is more or less "internal" to the implementation of the function. (I am not supper happy about this.)
Maybe it would just be better to drop the `--deduplicate` option. We are already building a Set that will handle deduplication and I don't think there is a perf concern. Worst case is the older client that gets two execs.
We could remove from cli case, too, for consistency. ?

Reply:
I pushed version that is much simpler. LMK if that looks better than the first draft - I'm good with either.